smart-contract/program/src/utils.rs

```diff
@@ -22,20 +22,29 @@ pub fn calc_reward_fp32(
     msg!("Nb of days behind {}", nb_days_to_claim);
     msg!("Last claimed offset {}", last_claimed_offset);
     msg!("Current offset {}", current_offset);
-    nb_days_to_claim = std::cmp::min(nb_days_to_claim, STAKE_BUFFER_LEN - 1);
+    nb_days_to_claim = std::cmp::min(nb_days_to_claim, STAKE_BUFFER_LEN);
+    msg!("Nb of days to claim {}", nb_days_to_claim);
+    if nb_days_to_claim == 0 {
+        if !allow_zero_rewards {
+            msg!("No rewards to claim, no operation.");
+            return Err(AccessError::NoOp.into());
+        }
+        return Ok(0);
+    }
 
     if current_offset > stake_pool.header.current_day_idx as u64 {
         #[cfg(not(any(feature = "days-to-sec-10s", feature = "days-to-sec-15m")))]
```

Comment:
Does it make sense to keep this features here?

Reply:
It doesn't make cause harm and I don't want to mix the feature removal with this PR